Hotfix for QC KeyError exception

awhoward opened the pull request:
Here's a PR to address the KeyError issue (example error statement below) that @RussLaher and @bjfultn mentioned during the DRP meeting today.

I added two things:
- KeyErrors are explicitly caught when running through all QCs.
- The method in Analyze2D that computes master ages now checks if the relevant keyword exists before reading it.  If it does not exist, it returns a value of -999 days for the age.

I have done some testing (running the Analyze2D.measure_master_age() method manually and running a simple recipe to exercise the Diagnostics and QC frameworks), but it is not extensive.
```
[QualityControlFramework][INFO]:Running QC: D2_master_bias_age (OLDBIAS; Check master dark file age)
ERROR: Problem with determining age of BIASFILE: "Keyword 'BIASFILE' not found."
Traceback (most recent call last):
  File "/code/KPF-Pipeline/modules/quicklook/src/analyze_2d.py", line 107, in measure_master_age
    master_filename = self.header[kwd]
  File "/usr/local/lib/python3.6/site-packages/astropy/io/fits/header.py", line 156, in __getitem__
    card = self._cards[self._cardindex(key)]
  File "/usr/local/lib/python3.6/site-packages/astropy/io/fits/header.py", line 1716, in _cardindex
    raise KeyError(f"Keyword {keyword!r} not found.")
KeyError: "Keyword 'BIASFILE' not found."
[QualityControlFramework][INFO]:QC result: False (True = pass)
```

RussLaher commented:
Hi Andrew, I am testing your changes this morning.  I just noticed the following:
```
[pipeline_20250209.log][ERROR]:Age of masters for Bias/Dark/Flat not computed: name 'master_keyword' is not defined
Traceback (most recent call last):
  File "/code/KPF-Pipeline/modules/quicklook/src/diagnostics_framework.py", line 132, in _perform
    self.kpf_object = diagnostics.add_headers_masters_age_2D(self.kpf_object, logger=self.logger)
  File "/code/KPF-Pipeline/modules/quicklook/src/diagnostics.py", line 407, in add_headers_masters_age_2D
    logger.error(f"Problem with {master_keyword} age determination: Age of {master_file} compared to this file (whole days) = {new_keyword}")
NameError: name 'master_keyword' is not defined


```

Also, here is another:
```
[QualityControlFramework][INFO]:Running QC: D2_master_dark_age (OLDDARK; Check master dark file age)
INFO: Exception: '>' not supported between instances of 'datetime.timedelta' and 'int'
[QualityControlFramework][INFO]:QC result: False (True = pass)
```

awhoward commented:
Thanks, @RussLaher.  I'll take a look, likely not before tonight.

awhoward commented:
Changes:
- I think I fixed the issue with `NameError: name 'master_keyword' is not defined`
- I made a change that should fix `INFO: Exception: '>' not supported between instances of 'datetime.timedelta' and 'int'`.  This was due to returning a timedelta object instead of an integer number of days.
- For the error related to KP.20250209.03737.72, I changed the master_types dictionary entry for POS2DSNR.  This was needed.  It looks like the QC tests for OLDDARK and OLDFLAT also failed, but this shouldn't matter because the master_types dictionary entries for them are set to empty lists (i.e., ignored by the master creation code).

@RussLaher - you might give it a try again.

awhoward commented:
This is ready to merge.
